sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/DataSourceV2AnalysisSuite.scala

```diff
@@ -0,0 +1,411 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import java.util.Locale
+
+import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference, Cast, UpCast}
+import org.apache.spark.sql.catalyst.plans.logical.{AppendData, LeafNode, LogicalPlan, Project}
+import org.apache.spark.sql.types.{DoubleType, FloatType, StructField, StructType}
+
+case class TestRelation(output: Seq[AttributeReference]) extends LeafNode with NamedRelation {
+  override def name: String = "table-name"
+}
+
+class DataSourceV2AnalysisSuite extends AnalysisTest {
+  val table = TestRelation(StructType(Seq(
+    StructField("x", FloatType),
+    StructField("y", FloatType))).toAttributes)
+
+  val requiredTable = TestRelation(StructType(Seq(
+    StructField("x", FloatType, nullable = false),
+    StructField("y", FloatType, nullable = false))).toAttributes)
+
+  val widerTable = TestRelation(StructType(Seq(
+    StructField("x", DoubleType),
+    StructField("y", DoubleType))).toAttributes)
+
+  test("Append.byName: basic behavior") {
+    val query = TestRelation(table.schema.toAttributes)
+
+    val parsedPlan = AppendData.byName(table, query)
+
+    checkAnalysis(parsedPlan, parsedPlan)
+    assertResolved(parsedPlan)
+  }
+
+  test("Append.byName: does not match by position") {
```

Comment:
this test is by name.

Reply:
Yes, this is testing that the query that would work when matching by position fails when matching by name.